src/Microsoft.Health.Fhir.SqlServer/Features/Search/PrimaryKeyValue.cs

```diff
@@ -0,0 +1,15 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search
+{
+    /// <summary>
+    /// Represents a row in the Resource or search parameter tables by its primary key.
+    /// </summary>
+    internal record PrimaryKeyValue(short ResourceTypeId, long ResourceSurrogateId)
```

Comment:
>record [](http://example.com/codeflow?start=13&length=6)

C# 9! woohoo!

Reply:
I think the Expression classes could benefit from becoming records one day. In many cases, we just need to update one property, so `with` expressions could be quite nice.